test/torchaudio_unittest/example/emformer_rnnt/test_librispeech_lightning.py

```diff
@@ -80,28 +52,33 @@ def get_lightning_module():
 class TestLibriSpeechRNNTModule(TorchaudioTestCase):
     @classmethod
     def setUpClass(cls) -> None:
+        super().setUpClass()
         torch.random.manual_seed(31)
 
-    def test_training_step(self):
-        with get_lightning_module() as lightning_module:
-            train_dataloader = lightning_module.train_dataloader()
-            batch = next(iter(train_dataloader))
-            lightning_module.training_step(batch, 0)
-
-    def test_validation_step(self):
-        with get_lightning_module() as lightning_module:
-            val_dataloader = lightning_module.val_dataloader()
-            batch = next(iter(val_dataloader))
-            lightning_module.validation_step(batch, 0)
-
-    def test_test_step(self):
-        with get_lightning_module() as lightning_module:
-            test_dataloader = lightning_module.test_dataloader()
-            batch = next(iter(test_dataloader))
-            lightning_module.test_step(batch, 0)
-
-    def test_forward(self):
+    @parameterized.expand(
+        [
+            ("train",),
+            ("dev",),
+            ("test",),
+            ("forward",),
+        ]
+    )
+    def test_step(self, subset):
         with get_lightning_module() as lightning_module:
-            val_dataloader = lightning_module.val_dataloader()
-            batch = next(iter(val_dataloader))
-            lightning_module(batch)
+            if subset == "train":
+                dataloader = lightning_module.train_dataloader()
+                step = lightning_module.training_step
+            elif subset == "dev":
+                dataloader = lightning_module.val_dataloader()
+                step = lightning_module.validation_step
+            elif subset == "test":
+                dataloader = lightning_module.test_dataloader()
+                step = lightning_module.test_step
+            else:
+                dataloader = lightning_module.val_dataloader()
+                step = lightning_module
+            batch = next(iter(dataloader))
+            if subset == "forward":
+                step(batch)
+            else:
+                step(batch, 0)
```

Comment:
if we want to parameterize it this way, i'd forego the mapping from "subset", e.g.

```
@parameterized.expand(
    [
        ("training_step", "train_dataloader"),
        # etc
    ]
)
def test_step(step_fname, dataloader_fname):
    with get_lightning_module() as lightning_module:
        dataloader = getattr(lightning_module, dataloader_fname)()
        batch = next(iter(dataloader))
        getattr(lightning_module, step_fname)(batch, 0)
```